core/math/matrix3.cpp

```diff
@@ -436,15 +432,10 @@ Vector3 Basis::get_euler_yxz() const {
 
 	if (m12 < 1) {
 		if (m12 > -1) {
-			if (elements[1][0] == 0 && elements[0][1] == 0 && elements[2][2] < 0) { // use pure x rotation
-				real_t x = asin(-m12);
+			if (elements[1][0] == 0 && elements[0][1] == 0 && elements[0][2] == 0 && elements[2][0] == 0 && elements[0][0] == 1) { // use pure x rotation
+				real_t x = atan2(-m12, elements[1][1]);
```

Comment:
`x` is unused since you removed the chunk below, I guess it should be `euler.x`?